### DISCUSSION TITLE
unstable_batchedUpdates necessary when modifying state within subscription handler?

### DISCUSSION THREAD

**Discussion by ROTGP** [pmndrs/zustand#809] [Q&A] [answered]
- I have two separate stores
- In `store1`, I subscribe to changes, which are fired as a result of user interaction in a component
- In the handler for the subscription in `store1`, I call something like `store2.getState().doSomething()`

My question: in this situation - is it necessary to wrap the code in the subscription handler in `unstable_batchedUpdates`?  I see in the docs "Updating the state outside an event handler will force react to update the components synchronously, therefore adding the risk of encountering the zombie-child effect" - what does this mean exactly?  How do we know if we're updating state within an event handler?

**Comment by dai-shi** [comment, marked as answer]
I don't think you need anything in that situation.
Or, put it the other way, you don't need it unless you observe zombie children, which means you receive two inconsistent values from two sources (like one from hooks and the other from props.)
